.gitlab-ci.yml

```diff
@@ -153,14 +153,17 @@ test-on-netgear-rax40:
   variables:
     # TESTS_TO_RUN needs to be set by the user (or the pipeline schedule)
     GIT_CLONE_PATH: "/builds/prpl-foundation/prplMesh/"
+    DEVICE_UNDER_TEST: prplmesh
   script:
-      - /easymesh_cert/run_test_file.sh -o logs $TESTS_TO_RUN
+      - tools/deploy_ipk.sh prplmesh-rax40 tools/docker/builder/openwrt/*.ipk
+      - /easymesh_cert/run_test_file.sh -o logs -d $DEVICE_UNDER_TEST $TESTS_TO_RUN
   artifacts:
     paths:
       - logs
     when: always
   needs:
     - job: build-in-docker
+    - job: build-for-netgear-rax40
```

Comment:
I'm thinking it would be better to have a separate job per device. Something like:

```
.run-certification-tests:
  stage: test
  variables:
    # TESTS_TO_RUN needs to be set by the user (or the pipeline schedule)
    # DEVICE_UNDER_TEST needs to be set by the user or pipeline schedule
    GIT_CLONE_PATH: "/builds/prpl-foundation/prplMesh/"
  script:
      - tools/deploy_ipk.sh $DEVICE_UNDER_TEST build/$DEVICE_UNDER_TEST/*.ipk
      - /easymesh_cert/run_test_file.sh -o logs -d $DEVICE_UNDER_TEST $TESTS_TO_RUN
  artifacts:
    paths:
      - logs
    when: always
  tags:
    - certs-tests
  timeout: 24h
  rules:
    - if: '$TESTS_TO_RUN =~ /^$/'
       when: manual
    - if: '$SOME_VARIABLE_SET_BY_SCHEDULE = ON'

certification-tests-ubuntu:
  extends: .run-certification-tests
  variables:
    DEVICE_UNDER_TEST: prplmesh
  needs:
    - job: build-in-docker

certification-tests-rax40:
  extends: .run-certification-tests
  variables:
    DEVICE_UNDER_TEST: netgear-rax40
  needs:
    - job: build-for-netgear-rax40
```

It would be even better if we could do `job: build-for-$DEVICE_UNDER_TEST` but I don't know if that's possible.

The rules syntax is also a bit fuzzy so I'm not sure if it will work like that.

Reply:
I really love the idea of combining the scheduled and manual jobs!
Looking for TESTS_TO_RUN value is genius. I believe the schedules set it _before_ the rules are evaluated, while when we want to run them manually the rules have already been evaluated (in other words, it should do what we want).

It also makes sense to have 2 separate jobs for the 2 different devices, but as it is, I think it will always run the tests for both devices when 1 schedule is run, so we miss the possibility to run only for one DUT on schedule.

We need to find a way for the schedules to specify a list of DUTs to run with while keeping 1 manual job per device like here.

Reply:
I tried something on `dev/ci-multiple-duts-merge-scheduled-manual`, but for some reason Gitlab marks the pipeline as "blocked" when there are manual, not triggered, jobs and no longer as passing: https://gitlab.com/prpl-foundation/prplMesh/pipelines/113248562 

Also if we want to split the certification tests between smaller groups, I'm starting to think that it would be simpler to just generate a yaml.

I'll make the other suggested change and keep these in a separate issue so that we can already merge this one and unblock #746 .

Reply:
Opened https://github.com/prplfoundation/prplMesh/issues/759 to track this.